buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KotlinNativeTest.kt

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+package org.jetbrains.kotlin
+
+import groovy.lang.Closure
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+import org.gradle.language.base.plugins.LifecycleBasePlugin
+import org.gradle.process.ExecSpec
+
+import java.io.File
+import java.io.ByteArrayOutputStream
+import java.util.regex.Pattern
+
+import org.jetbrains.kotlin.konan.target.HostManager
+
+abstract class KonanTest : DefaultTask() {
+    enum class Logger {
+        EMPTY,    // Built without test runner
+        GTEST,    // Google test log output
+        TEAMCITY, // TeamCity log output
+        SIMPLE,   // Prints simple messages of passed/failed tests
+        SILENT    // Prints no log of passed/failed tests
+    }
+
+    var disabled: Boolean
+        get() = !enabled
+        @Optional set(value) { enabled = !value }
+
+    /**
+     * Test output directory. Used to store processed sources and binary artifacts.
+     */
+    lateinit var outputDirectory: String
```

Comment:
Can we make `outputDirectory`, `testLogger` and `executable` abstract and get rid of `lateinits`? We still initialize these properties in constructors of subclasses.

Reply:
Removed most of lateinits.